drivers/sensor/tdk/icm42x70/icm42670.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2025 TDK Invensense
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "icm42670.h"
+
+#include <zephyr/drivers/sensor/icm42x70.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ICM42670, CONFIG_SENSOR_LOG_LEVEL);
+
+static uint32_t convert_gyr_fs_to_bitfield(uint32_t val, uint16_t *fs)
+{
+	uint32_t bitfield = 0;
+
+	if (val < 500 && val >= 250) {
+		bitfield = GYRO_CONFIG0_FS_SEL_250dps;
+		*fs = 250;
+	} else if (val < 1000 && val >= 500) {
+		bitfield = GYRO_CONFIG0_FS_SEL_500dps;
+		*fs = 500;
+	} else if (val < 2000 && val >= 1000) {
+		bitfield = GYRO_CONFIG0_FS_SEL_1000dps;
+		*fs = 1000;
+	} else if (val == 2000) {
+		bitfield = GYRO_CONFIG0_FS_SEL_2000dps;
+		*fs = 2000;
+	}
+	return bitfield;
+}
+
+static int icm42670_set_gyro_odr(struct icm42x70_data *drv_data, const struct sensor_value *val)
+{
+	if (val->val1 <= 1600 && val->val1 > 12) {
+		if (drv_data->gyro_hz == 0) {
+			inv_imu_set_gyro_frequency(
+				&drv_data->driver,
+				convert_freq_to_bitfield(val->val1, &drv_data->gyro_hz));
+			inv_imu_enable_gyro_low_noise_mode(&drv_data->driver);
+		} else {
+			inv_imu_set_gyro_frequency(
+				&drv_data->driver,
+				convert_freq_to_bitfield(val->val1, &drv_data->gyro_hz));
+		}
+	} else if (val->val1 == 0) {
+		inv_imu_disable_gyro(&drv_data->driver);
+		drv_data->gyro_hz = val->val1;
+	} else {
+		LOG_ERR("Incorrect sampling value");
+		return -EINVAL;
+	}
+	return 0;
+}
+
+static int icm42670_set_gyro_fs(struct icm42x70_data *drv_data, const struct sensor_value *val)
+{
+	if (val->val1 > 2000 || val->val1 < 250) {
+		LOG_ERR("Incorrect fullscale value");
+		return -EINVAL;
```

Comment:
`/** Sensor range, in SI units. */`

2000 and 250 are in degrees. You should expect the value in radians.

Reply:
It's a bug present in the current driver so thanks for this review. This is fixed now.